### ISSUE TITLE
question: securing ~$HOME/.gdfuse/default/state

### ISSUE THREAD

**Issue by mcrucifix** [astrada/google-drive-ocamlfuse#203]
I apologize in advance if this question betrays my mis-understanding of OAuth2 authentification... but: 

is this correct that someone who would have access to the information in  ~$HOME/.gdfuse/default/state could impersonate you ? If yes, isnt't it worth securing this information better, e.g., by using applications like gnome keyring (or keychain access on mac ?).

**Comment by astrada** [comment]
Actually, you need client id, client secret, **and** a refresh token (that is stored in ~$HOME/.gdfuse/default/state) to access someone's Drive account. So, knowing only your refresh token is not enough. The other access token that is stored in that file would be enough, but it's short lived (1 hour).

**Comment by mcrucifix** [comment]
mmm.... someone how would steal the computer and access the physical drive would be able to keep getting refreshing tokens, isnt'it ? . If course this won't happen if the gdfuse config dir is on an encrypted section.

**Comment by astrada** [comment]
Yes, but you could revoke the token from https://myaccount.google.com/security#connectedapps

**Comment by R-Chandra** [comment]
You could always use something like ecryptfs to encrypt the contents of $HOME/.gdfuse if you wanted to secure it better.  That would require, at minimum, inserting the proper crypto keys into the kernel in order to read the data (which is done by the utilities in the ecryptfs-utils Ubuntu package, such as ecryptfs-mount-private).  I haven't delved deeply into that, but by default it encrypts into $HOME/.Private and mounted on $HOME/Private.  I have done the setup and then edited $HOME/.ecryptfs/Private.mnt to mount it on $HOME/encry, so it could equally do .gdfuse.  Presumably you could also set up more than one of these, just .Private and Private being the defaults.

**Comment by mcrucifix** [comment]
I agree that this is a reasonable solution. 

Michel.

**Comment by scoopydude2002** [comment]
I only use whole disk encryption on all my Linux servers now. Easy to set up and performance is better.
